src/collections/infra/repositories/CollectionsRepository.ts

```diff
@@ -0,0 +1,25 @@
+import { ApiRepository } from '../../../core/infra/repositories/ApiRepository'
+import { ICollectionsRepository } from '../../domain/repositories/ICollectionsRepository'
+import { transformCollectionIdResponseToPayload } from './transformers/collectionTransformers'
+import { Collection } from '../../domain/models/Collection'
+export class CollectionsRepository extends ApiRepository implements ICollectionsRepository {
+  private readonly collectionsResourceName: string = 'dataverses'
+  private readonly collectionsDefaultOperationType: string = 'get'
+
+  public async getCollection(
+    collectionObjectParameter: number | string = 'root'
+  ): Promise<Collection> {
+    return this.doGet(
+      this.buildApiEndpoint(
+        this.collectionsResourceName,
+        this.collectionsDefaultOperationType,
+        collectionObjectParameter
+      ),
+      true
+    )
+      .then((response) => transformCollectionIdResponseToPayload(response))
+      .catch((error) => {
+        throw error
+      })
+  }
+}
```

Comment:
I forgot about this when we were creating the issue, but we also need the owner's information for the collection.

From the main repository documentation:

```
If you want to include the Dataverse collections that this collection is part of, you must set the `returnOwners` query parameter to `true`.

Usage example:

.. code-block:: bash
   curl "https://demo.dataverse.org/api/dataverses/root?returnOwners=true"
```

Suggested implementation:

```suggestion
  public async getCollection(
    collectionObjectParameter: number | string = 'root'
  ): Promise<Collection> {
    return this.doGet(
      this.buildApiEndpoint(
        this.collectionsResourceName,
        this.collectionsDefaultOperationType,
        collectionObjectParameter
      ),
      true,
      { returnOwners: true }
    )
      .then((response) => transformCollectionIdResponseToPayload(response))
      .catch((error) => {
        throw error;
      });
  }
}
```

Please, check the implementation of `getDataset` to see how the property `isPartOf` it's implemented in the transformer @M27Mangan